### ISSUE TITLE
Legends inside and outside of the plot

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaPlots/Makie.jl#61]
Common feature request for Plots.jl. It should be planned for early.

**Comment by mkborregaard** [comment]
Could be implemented by making the legend a subplot in itself that can be placed either next to or superimposed to existing subplots - would be great for a shared legend for multi-subplot figures.

**Comment by SimonDanisch** [comment]
You can place the legend wherever you want already: http://makie.juliaplots.org/stable/legends.html

**Comment by ChrisRackauckas** [comment]
It might be nice in the simple functions to make that an attribute.

**Comment by mkborregaard** [comment]
(offtopic - I can't see the video on that page on my mac - firefox/safari)

**Comment by dlfivefifty** [comment]
The link http://makie.juliaplots.org/stable/legends.html no longer works.

I'm trying to figure out how to add a legend to explain the colours for a 2D contour (mesh) plot.

**Comment by SimonDanisch** [comment]
Sorry, I just added examples from home, but didn't push them :D will push them this evening!

**Comment by SimonDanisch** [comment]
Ah, just found one of the examples on my work PC:
```Julia
using Makie
scene = Scene(resolution = (500, 500))
x = map([:dot, :dash, :dashdot], [2, 3, 4]) do ls, lw
    linesegments!(
        range(1, stop = 5, length = 100), rand(100), rand(100), 
        linestyle = ls, linewidth = lw,
        color = rand(RGBAf0)
    )[end]
end
x = [x..., scatter!(range(1, stop=5, length=100), rand(100), rand(100))[end]]
center!(scene)
ls = Makie.legend(x, ["attribute $i" for i in 1:4], camera = campixel!, raw = true)
vbox(scene, ls)
```
![image](https://user-images.githubusercontent.com/1010467/48356413-7fe50b00-e696-11e8-955f-9f27a03f5c3d.png)

**Comment by grero** [comment]
Cool, thanks! That works for me : )

**Comment by grero** [comment]
Just a follow up, how would I go about creating a legend inside an existing plot? What I have attempted so far is to initially place the legend and plot side-by-side using the code above, and then using Makie.translate! on the plot to move it inside the plot. I encountered two issues with this. One is that the legend moves behind the plot instead, and the second is that the bounding box does not seem to update to reflect the fact that the legend has moved. Is there a recommended way of placing the legend inside the plot? Thanks!
